torch/distributed/distributed_c10d.py

```diff
@@ -1704,11 +1705,15 @@ def _new_process_group_helper(
             return GroupMember.NON_GROUP_MEMBER, None
 
     prefix_store = PrefixStore(f"{group_name}/", store)
-    base_pg_options = ProcessGroup.Options(backend=str(backend))
-    base_pg_options._timeout = timeout
+    # The backend for PG will be set later based on what's inside BackendConfig
+    # and timeout are set in each backend's option.
     pg: ProcessGroup = ProcessGroup(
-        prefix_store, group_rank, group_size, base_pg_options
+        prefix_store,
+        group_rank,
+        group_size,
     )
+    assert backend in Backend.backend_type_map, f"Unknown backend type {backend}"
+    pg._set_default_backend(Backend.backend_type_map[backend])
```

Comment:
if backend isn't provided (e.g. `backend=None`), then the backend type will be undefined right? Will this break the APIs that depend on the looking at the backend type, like getSequenceNumber or the hooks that we talked about.

I'm not sure if we have tests that cover these cases, but could be worthwhile to check.

Reply:
The backend is not optional for this function. Do we see the use case when `backend=None`? This logic will break when the backend is something like "cpu:gloo,gpu:nccl".